compiler_opt/rl/feature_ops_test.py

```diff
@@ -41,6 +43,36 @@ def setUp(self):
     self._quantile_file_dir = os.path.join(constant.BASE_DIR, 'testdata')
     super().setUp()
 
+  def _create_temp_vocab_file(self, vocab_data):
+    """Helper method to create a temporary vocab file with given data."""
+    with tempfile.NamedTemporaryFile(
+        mode='w', suffix='.json', delete=False) as f:
+      if isinstance(vocab_data, str):
+        f.write(vocab_data)  # For invalid JSON
+      else:
+        json.dump(vocab_data, f)
+      return f.name
+
+  def _test_vocab_file_dimensions(self, vocab_data, expected_dimensions):
+    """Helper method to test vocab file dimension reading."""
+    temp_file = self._create_temp_vocab_file(vocab_data)
+    try:
```

Comment:
Here and below in `_test_vocab_file_dimensions_with_exception`:

If you refactor the temp vocab file to be created in the test, you can then drop the try/finally here.

Reply:
You want me to create temp file locally within each test?

Reply:
Just within the `_test_vocab_file_*` functions.